### PR TITLE
Update Solr::Query::HttpRequestBuilder to accept offset/rows arguments

### PR DESCRIPTION
**Pull request by drnluz**

### CODE REVIEW

File: lib/solr/query/request.rb [hunk 1]
```diff
@@ -32,9 +32,9 @@ def initialize(search_term:, fields: [], filters: [])
 
       # Runs this Solr::Request against Solr and
       # returns [Solr::Response]
-      def run(page: 1, page_size: 10)
+      def run(page: nil, page_size: nil, offset: nil, rows: nil)
```
Comment (Valve) [inline lib/solr/query/request.rb]: what if someone specifies all options? Are they not mutually exclusive?

Comment (Valve) [reply lib/solr/query/request.rb]: Also solr calls these `start` and `rows`. 
So I would vote to just have these 2 params as defaults, but move page/gize size into a `run_paged` or something (for example).

Comment (drnluz) [reply lib/solr/query/request.rb]: good catch, `rows` and `page_size` are the exact same thing.
Let me reorganize this class, it's not good the way it is now.

**Comment by Valve** [comment]
@drnluz do we still need this PR?

**Comment by drnluz** [comment]
@Valve Not anymore, closing it.

**Pull request closed without merging**